tests/api/helpers.rs

```diff
@@ -64,12 +105,21 @@ pub async fn spawn_app() -> TestApp {
         .expect("Failed to build application");
 
     let address = format!("http://{}", server.local_addr());
+    let port = address
```

Comment:
Get the port from the generated address for testing. In prod, the base url sent in the confirmation email doesnt need the port of because of DNS lookup, but when testing we do need the port